bin/proving-service/src/proxy/status.rs

```diff
@@ -24,3 +37,96 @@ pub async fn create_status_client(
 
     Ok(StatusApiClient::new(channel))
 }
+
+/// Status of a worker
+#[derive(Debug, Serialize, Deserialize)]
+pub struct WorkerStatus {
+    address: String,
+    is_available: bool,
+    is_healthy: bool,
+    retries: usize,
+}
+
+/// Status of the proxy
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ProxyStatus {
+    version: String,
+    supported_proof_types: ProverTypeSupport,
+    total_workers: usize,
+    busy_workers: usize,
+    workers: Vec<WorkerStatus>,
+}
+
+/// Service that handles status requests
+pub struct StatusService {
+    load_balancer: Arc<LoadBalancerState>,
+}
+
+impl StatusService {
+    pub fn new(load_balancer: Arc<LoadBalancerState>) -> Self {
+        Self { load_balancer }
+    }
+
+    async fn handle_request(&self, _session: &mut ServerSession) -> Result<()> {
+        let workers = self.load_balancer.workers.read().await;
```

Comment:
Looks like this lock could be dropped earlier to reduce contention

Reply:
Also, because this locks potential writes, does this introduce a potential DDoS vector if users send a bunch of status queries? Although this depends on whether this is user facing as well. An alternate strategy could be to read the status internally every minute or so and then respond with the cached response. Or maybe when the health checks are performed, the results are saved on another struct which does not lock the workers list.

EDIT: Just read on the issue that initially it was mostly for internal/semi-internal users. If so then maybe we could disregard this but it's something that we want to keep in mind or change in a follow-up PR

Reply:
I like the idea of reducing DoS attack vectors as well, but I'd probably do this as a separate issue.